internal/backend/remote-state/s3/client.go

```diff
@@ -109,6 +112,30 @@ func (c *RemoteClient) get(ctx context.Context) (*remote.Payload, error) {
 	var output *s3.GetObjectOutput
 	var err error
 
+	ctx, baselog := baselogging.NewHcLogger(ctx, logging.HCLogger().Named("backend-s3"))
+	ctx = baselogging.RegisterLogger(ctx, baselog)
+
+	inputHead := &s3.HeadObjectInput{
+		Bucket: &c.bucketName,
+		Key:    &c.path,
+	}
+
+	// Head works around some s3 compatible backends not handling missing GetObject requests correctly (ex: minio Get returns Missing Bucket)
+	_, err = c.s3Client.HeadObject(ctx, inputHead)
+	if err != nil {
+		var nb *types.NoSuchBucket
+		if errors.As(err, &nb) {
+			return nil, fmt.Errorf(errS3NoSuchBucket, err)
+		}
+
+		var nk *types.NotFound
+		if errors.As(err, &nk) {
+			return nil, nil
+		}
+
+		return nil, err
+	}
```

Comment:
**(nitpick)** It may be a matter of taste, but I'd like to suggest the following refactoring 😃 

```suggestion
	// Head works around some s3 compatible backends not handling missing GetObject requests correctly (ex: minio Get returns Missing Bucket)
	_, err = c.s3Client.HeadObject(ctx, &s3.HeadObjectInput{
		Bucket: &c.bucketName,
		Key:    &c.path,
	})
	
	switch err.(type) {
	case nil:
	case *types.NotFound:
		return nil, nil
	case *types.NoSuchBucket:
		return nil, fmt.Errorf(errS3NoSuchBucket, err)
	default:
		return nil, err
	}
```

Reply:
I tried to match the current style, IMO there's a log of potential clean up that could happen in this backend.  The question is how much to do now and how much might be broken during a refactor.

Reply:
I'm personally not a huge fan of handling errors in this way, especially as we're only trying to check for 2 types. I'm happy with the handling as it is now.

Reply:
I'm also 50/50 on inlining the parameter of the headObjectInput. Feel free to go with whichever you guys prefer 👍

Reply:
>I'm personally not a huge fan of handling errors in this way

Yes. I agree.

My suggestion would be to just abstract those if checks in to a function `func sanitizeErr(err error) err` or something with a better name. Check and convert the error there and reuse it in other places.

Reply:
2 vs. 1 - democracy wins 😄